phoenix-core/src/it/java/org/apache/phoenix/end2end/AlterAddCascadeIndexIT.java

```diff
@@ -0,0 +1,339 @@
+import static org.junit.Assert.assertTrue;
+
+@RunWith(Parameterized.class)
+public class AlterAddCascadeIndexIT extends ParallelStatsDisabledIT {
+
+    public static final String SYNTAX_ERROR = "Syntax error";
+    @Rule
+    public ExpectedException exception = ExpectedException.none();
+    private static Connection conn;
+    private Properties prop;
+    private boolean isViewIndex;
+    private String phoenixObjectName;
+    private String indexesName;
+    private final String tableDDLOptions;
+
+
+    public AlterAddCascadeIndexIT(boolean isViewIndex, boolean mutable) {
+        this.isViewIndex = isViewIndex;
+        StringBuilder optionBuilder = new StringBuilder();
+        if (!mutable) {
+            optionBuilder.append(" IMMUTABLE_ROWS=true");
+        }
+        this.tableDDLOptions = optionBuilder.toString();
+    }
+
+    @Parameters(name="AlterAddCascadeIndexIT_isViewIndex={0},mutable={1}")
+    public static Collection<Object[]> data() {
+        return Arrays.asList(new Object[][] {
+                { true, true},
+                { true, false},
+                { false, true},
+                { false, false},
+        });
+    }
+
+    @Before
+    public void setup() throws SQLException {
+        prop = new Properties();
+        conn = DriverManager.getConnection(getUrl(), prop);
+        conn.setAutoCommit(true);
+        conn.createStatement().execute("CREATE TABLE IF NOT EXISTS test.us_population (\n" +
+                "      state CHAR(2) NOT NULL,\n" +
+                "      city VARCHAR NOT NULL,\n" +
+                "      population BIGINT,\n" +
+                "      CONSTRAINT my_pk PRIMARY KEY (state, city)) " + tableDDLOptions);
+
+        if(isViewIndex) {
+            conn.createStatement().execute("CREATE VIEW IF NOT EXISTS test.us_population_gv" +
+                    "(city_area INTEGER, avg_fam_size INTEGER) AS " +
+                    "SELECT * FROM test.us_population WHERE state = 'CA'");
+
+            conn.createStatement().execute("CREATE INDEX IF NOT EXISTS us_population_gv_gi ON " +
+                    "test.us_population_gv (city_area) INCLUDE (population)");
+            conn.createStatement().execute("CREATE INDEX IF NOT EXISTS us_population_gv_gi_2 ON " +
+                    "test.us_population_gv (avg_fam_size) INCLUDE (population)");
+            phoenixObjectName = "test.us_population_gv";
+            indexesName = "us_population_gv_gi, us_population_gv_gi_2";
+        } else {
+            conn.createStatement().execute("CREATE INDEX IF NOT EXISTS us_population_gi ON " +
+                    "test.us_population (population)");
+            conn.createStatement().execute("CREATE INDEX IF NOT EXISTS us_population_gi_2 ON " +
+                    "test.us_population (state, population)");
+            phoenixObjectName = "test.us_population";
+            indexesName = "us_population_gi, us_population_gi_2";
+        }
+    }
+
+    // Test with ALTER TABLE CASCADE INDEX ALL
```

Comment:
Nit: The comment here and line 134 are misleading. The tests are handling both tables and views.